File: 프로그래머스/2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string s) {
    string answer = "";
    bool flag = true;
    for (char c:s){
        if (c == ' '){
            flag = true;
            answer += c;
        }
        else{
            if (flag == true){
                flag = false;
                answer += toupper(c);
            }
            else{
                answer += tolower(c);
            }
        }
    }
    return answer;
}

File: 프로그래머스/2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.py
def solution(s):
    answer = ''
    flag = True
    for c in s:
        if c == ' ':
                flag = True
                answer += c
        elif flag:  
            if c.isalpha():
                answer += c.upper()
            else:
                answer += c
            flag = False
        else:
            if c == ' ':
                flag = True
                answer += c
            elif c.isalpha():
                answer += c.lower()
            else:
                answer += c
        
    return answer